src/store/base.rs

```diff
@@ -0,0 +1,701 @@
+//! Abstraction over the caching implementation
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+use error_stack::{IntoReport, Report, Result, ResultExt};
```

Comment:
```suggestion
use error_stack::{ensure, report, IntoReport, Result, ResultExt};
```


Use [ensure!](https://docs.rs/error-stack/latest/error_stack/macro.ensure.html) and [report!](https://docs.rs/error-stack/latest/error_stack/macro.report.html) macros for less boilerplate code.